packages/eslint-plugin/tests/areOptionsValid.ts

```diff
@@ -0,0 +1,43 @@
+import type { RuleModule } from '@typescript-eslint/utils/ts-eslint';
+import Ajv from 'ajv';
+import type { JSONSchema4 } from 'json-schema';
+
+const ajv = new Ajv({ async: false });
+
+export function areOptionsValid(
+  rule: RuleModule<string, unknown[]>,
+  options: unknown,
+): boolean {
+  const normalizedSchema = normalizeSchema(rule.meta.schema);
+
+  const valid = ajv.validate(normalizedSchema, options);
+  if (typeof valid !== 'boolean') {
+    // Schema could not validate options synchronously. This is not allowed for ESLint rules.
+    return false;
+  }
+
+  return valid;
+}
+
+function normalizeSchema(
+  schema: JSONSchema4 | readonly JSONSchema4[],
+): JSONSchema4 {
+  if (!Array.isArray(schema)) {
```

Comment:
We could export this from the `index.ts` in the package and use it here
https://github.com/typescript-eslint/typescript-eslint/blob/3e3b7891a73136136929b6912ace9301b30fb0f3/packages/rule-schema-to-typescript-types/src/isArray.ts

Which will fix the safety caused by the TS bug

Reply:
Nice - didn't see this helper existed. Have moved it from the rule-schema-to-typescript-types package to the utils package and have used here.